effekt/shared/src/main/scala/effekt/generator/llvm/PrettyPrinter.scala

```diff
@@ -66,8 +66,12 @@ ${indentedLines(instructions.map(show).mkString("\n"))}
     case BitCast(result, operand, tpe) =>
       s"${localName(result)} = bitcast ${show(operand)} to ${show(tpe)}"
 
+    // TODO: Require `operand0` to be of integer type. It is currently hackily typed dependent on its right operand's type which ought to be a constant.
     case Add(result, operand0, ConstantInt(n)) =>
       s"${localName(result)} = add ${show(operand0)}, $n"
+    // TODO: Require `operand0` to be of floating-point type. It is currently hackily typed dependent on its right operand's type which ought to be a constant.
+    case Add(result, operand0, ConstantDouble(x)) =>
```

Comment:
Add a new instruction `FAdd` instead of reusing `Add` for floating point.

Reply:
I surmise https://github.com/effekt-lang/effekt/pull/141/commits/17c4807a0e308f4ecd31fcfe7bfed99d2457c195 is sufficient.